### PR TITLE
Throw away baselines

### PR DESCRIPTION
**Pull request by aewallwi**
A convenient script to throw away all flagged antennas. Adding this to the beginning of RTP stage 2 can save time and space. Depends on #714

### CODE REVIEW

File: scripts/throw_away_flagged_antennas.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+"""Command-line script for throw away all data with apriori flags. Saves downstream I/O, memory etc..."""
+
+import sys
+import argparse
+from hera_cal import io
+from hera_qm import utils
+
+# Parse arguments
+ap = argparse.ArgumentParser(description="Completely Flag a data file.")
```
Comment (jsdillon) [inline scripts/throw_away_flagged_antennas.py:14]: Docstring is wrong.

File: scripts/throw_away_flagged_antennas.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ap.add_argument("infilename", type=str, help="path to visibility data to completely flag.")
+ap.add_argument("outfilename", type=str, help="path to new visibility file to write out completely flagged data")
+ap.add_argument("--yaml_file", default=False, action="store_true", help='overwrites existing file at outfile')
```
Comment (jsdillon) [inline scripts/throw_away_flagged_antennas.py:17]: Docstring is wrong

**Review by jsdillon — CHANGES_REQUESTED** [review]
Couple of docstring problems.

Also, do you want this script to also throw away any baselines that are completely flagged already? I feel like that would be a nice, useful feature.

File: scripts/throw_away_flagged_antennas.py (hunk 2 above)
Comment (aewallwi) [reply scripts/throw_away_flagged_antennas.py]: fixed.

File: scripts/throw_away_flagged_antennas.py (hunk 1 above)
Comment (aewallwi) [reply scripts/throw_away_flagged_antennas.py]: fixed.

**Comment by aewallwi** [comment]
> Couple of docstring problems.
> 
> Also, do you want this script to also throw away any baselines that are completely flagged already? I feel like that would be a nice, useful feature.

I agree. 

This was a bit more then could be put in a shallow script so I've moved the baseline throwing away function into `io` with some unittests.

File: hera_cal/io.py [hunk 3]
```diff
@@ -1829,3 +1830,63 @@ def baselines_from_filelist_position(filename, filelist):
...
+    if throw_away_fully_flagged_data_baselines:
+        antpairs_to_keep = []
+        for antpair in hd.get_antpairs():
+            fully_flagged = True
+            for pol in hd.pols:
+                fully_flagged = fully_flagged & np.all(hd.get_flags(antpair + (pol, )))
+            if not fully_flagged:
+                antpairs_to_keep.append(antpair)
+        hd.select(bls=antpairs_to_keep)
+    hd.write_uvh5(outfilename, clobber=clobber)
```
Comment (jsdillon) [inline hera_cal/io.py:1880]: Can I suggest adding something to the history explaining which baselines were thrown out?

File: hera_cal/io.py [hunk 4]
```diff
@@ -1829,3 +1830,63 @@ def baselines_from_filelist_position(filename, filelist):
...
+
+
+def throw_away_flagged_ants_parser():
+    # Parse arguments
+    ap = argparse.ArgumentParser(description="Throw away baselines whose antennas are flagged in a yaml file or which have all integrations/chans flagged.")
+    ap.add_argument("infilename", type=str, help="path to visibility data to completely flag.")
```
Comment (jsdillon) [inline hera_cal/io.py:1886]: Wrong docstirng

File: hera_cal/io.py [hunk 5]
```diff
@@ -1829,3 +1830,63 @@ def baselines_from_filelist_position(filename, filelist):
...
+    ap.add_argument("outfilename", type=str, help="path to new visibility file to write out completely flagged data.")
```
Comment (jsdillon) [inline hera_cal/io.py:1887]: Wrong docstring

File: hera_cal/io.py [hunk 6]
```diff
@@ -1829,3 +1830,63 @@ def baselines_from_filelist_position(filename, filelist):
...
+            if True, also throw away baselines where all data is flagged.
+            Warning: Don't use this for files with a small number of time integrations.
+                     since this can easily happen by chance in files like this.
+            Default is False
+        clobber: bool, optional
+            overwrite output file if it already exists.
+            Default is False.
+    Returns
+    -------
+        N/A
```
Comment (jsdillon) [inline hera_cal/io.py:1857]: I'd suggest returning `hd`

**Review by jsdillon — CHANGES_REQUESTED** [review]
Looks good. Got a few suggestions/comments

File: hera_cal/io.py (hunk 5 above)
Comment (aewallwi) [reply hera_cal/io.py]: fixed.

File: hera_cal/io.py (hunk 4 above)
Comment (aewallwi) [reply hera_cal/io.py]: fixed.

File: hera_cal/io.py (hunk 3 above)
Comment (aewallwi) [reply hera_cal/io.py:1890]: added history string.

File: hera_cal/io.py (hunk 6 above)
Comment (aewallwi) [reply hera_cal/io.py:1858]: I've added this.

**Review by jsdillon — APPROVED** [review]

**Review by jsdillon — APPROVED** [review]
Approved once you fix the test passing issue, which I think is just `from . import version` rather than `import version`.

**Pull request merged**